util/tracing/detect/recorder.go

```diff
@@ -99,17 +158,22 @@ func (r *TraceRecorder) ExportSpans(ctx context.Context, spans []sdktrace.ReadOn
 		v.last = now
 		v.spans = append(v.spans, ss)
 	}
-	r.mu.Unlock()
-
-	if r.SpanExporter == nil {
-		return nil
-	}
-	return r.SpanExporter.ExportSpans(ctx, spans)
+	return nil
 }
 
 func (r *TraceRecorder) Shutdown(ctx context.Context) error {
-	if r.SpanExporter == nil {
+	// Initiate the shutdown of the gc loop.
+	r.shutdown(context.Canceled)
```

Comment:
add `errors.WithStack()`

Reply:
Deja vu github? If there's an area where I've missed adding `errors.WithStack` please tell me, but github seems to be showing me the same comment twice?